drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1670 @@
+  elseif ib.status == DoorLock.types.DlStatus.INVALID_FIELD then
+    status = "invalidCommand"
+  end
+
+  -- Update User in table
+  if status == "success" then
+    if cmdName == "addUser" then
+      add_user_to_table(device, userIdx, userType)
+    elseif cmdName == "updateUser" then
+      update_user_in_table(device, userIdx, userType)
+    end
+  else
+    device.log.warn(string.format("Failed to set user: %s", status))
+  end
+
+  -- Update commandResult
+  local result = {
+    commandName = cmdName,
+    userIndex = userIdx,
+    statusCode = status
+  }
+  local event = capabilities.lockUsers.commandResult(
+    result,
+    {
+      state_change = true,
+      visibility = {displayed = false}
+    }
+  )
+  device:emit_event(event)
+  device:set_field(lock_utils.BUSY_STATE, false, {persist = true})
+end
+
+-----------------
+-- Delete User --
+-----------------
+local function handle_delete_user(driver, device, command)
+  -- Get parameters
+  local cmdName = "deleteUser"
+  local userIdx = command.args.userIndex
+
+  -- Check busy state
+  local busy = check_busy_state(device)
+  if busy == true then
+    local result = {
+      commandName = cmdName,
+      statusCode = "busy"
+    }
+    local event = capabilities.lockUsers.commandResult(
+      result,
+      {
+        state_change = true,
+        visibility = {displayed = false}
+      }
+    )
+    device:emit_event(event)
+    return
+  end
+
+  -- Save values to field
+  device:set_field(lock_utils.COMMAND_NAME, cmdName, {persist = true})
+  device:set_field(lock_utils.USER_INDEX, userIdx, {persist = true})
+
+  -- Send command
+  local ep = device:component_to_endpoint(command.component)
+  device:send(DoorLock.server.commands.ClearUser(device, ep, userIdx))
+end
+
+----------------------
+-- Delete All Users --
+----------------------
+local function handle_delete_all_users(driver, device, command)
+  -- Get parameters
+  local cmdName = "deleteAllUsers"
+
+  -- Check busy state
+  local busy = check_busy_state(device)
+  if busy == true then
+    local result = {
+      commandName = cmdName,
+      statusCode = "busy"
+    }
+    local event = capabilities.lockUsers.commandResult(
+      result,
+      {
+        state_change = true,
+        visibility = {displayed = false}
+      }
+    )
+    device:emit_event(event)
+    return
+  end
+
+  -- Save values to field
+  device:set_field(lock_utils.COMMAND_NAME, cmdName, {persist = true})
+  device:set_field(lock_utils.USER_INDEX, ALL_INDEX, {persist = true})
+
+  -- Send command
+  local ep = device:component_to_endpoint(command.component)
+  device:send(DoorLock.server.commands.ClearUser(device, ep, ALL_INDEX))
+end
+
+-------------------------
+-- Clear User Response --
+-------------------------
+local function clear_user_response_handler(driver, device, ib, response)
+  -- Get result
+  local cmdName = device:get_field(lock_utils.COMMAND_NAME)
+  local userIdx = device:get_field(lock_utils.USER_INDEX)
```

Comment:
I think you should use the userIdx provided in the clearUser command args instead of this device field. https://librarian-regionals.smartthingsgdev.com/matter/Matter_1-3/appclusters.html#_clearuser_command